website/content/en/docs/about/under-the-hood/guarantees.md

```diff
@@ -11,6 +11,33 @@ Here you can find an overview of delivery guarantee types and their meaning as w
 the stability of our components. Next, you can head over to the [components] page and use filters to
 see which components support specific guarantees.
 
+## Acknowledgement guarantees
+
+Vector supports end-to-end acknowledgement for the majority of its
+sources and sinks. This is a system which tracks the delivery status of
+an event through the lifetime of that event as it travels from the
+originating source to any number of destination sinks. Support for this
+feature is indicated by the "acknowledgements" badge at the top of the
+relevant sink or source configuration reference page.
+
+For a source that supports end-to-end acknowledgements and has the
+`acknowledgements` option enabled, this will cause it to wait for _all_
+connected sinks to either mark the event as delivered or to persist the
+events to a durable buffer before acknowledging receipt of the event. If
+the sink signals the event was rejected and the source can provide an
+error response (ie through an HTTP error code), then the source will
+provide an appropriate error to the sending agent.
+
+Sinks support end-to-end acknowledgements by providing an indicator of
+the final status of each event after delivery has completed. This
+includes waiting until all internal buffering and the retry process is
+complete. Sinks which have a durable buffer configured will mark events
+as delivered once they are persisted to that buffer, as an indicator
+that Vector will continue to retry the event even after restarts. If the
+sink does not support acknowledgements, events will be marked as having
+been delivered when they are handed off to the sink component, before
+any deliveries are attempted.
+
```

Comment:
One question that came up recently, that we may want to get ahead of by documenting here, is what happens if events are dropped somewhere in the topology? My understanding is that they will be "acknowledged" by the source.

Reply:
Correct. I will add a note about that.